DSCResources/ProcessMitigation/ProcessMitigation.psm1

```diff
@@ -146,78 +187,418 @@ function Test-TargetResource
         [string]
         $MitigationTarget,
 
-        [Parameter()]
-        [string[]]
-        $Enable,
+        [Parameter(Mandatory = $true)]
+        [string]
+        $MitigationType,
+
+        [Parameter(Mandatory = $true)]
+        [string]
+        $MitigationName,
 
-        [Parameter()]
-        [string[]]
-        $Disable
+        [Parameter(Mandatory = $true)]
+        [string]
+        $MitigationValue
     )
 
     $inDesiredState = $true
     $currentState = Get-TargetResource @PSBoundParameters
 
-    # verify policies in Enable are in a desired state
-    foreach ($policy in $Enable)
+    if ($mitigationTarget -eq "System")
     {
-        if ($policy -notin $currentState.Enable)
+        if ($currentState.MitigationPolicy.SystemConfig.$MitigationType.$mitigationName -ne $mitigationValue)
         {
-            Write-Verbose -Message ($script:localizedData.policyNotInDesiredStateEnabled -f $policy)
+            Write-Verbose -Message ($script:localizedData.policyNotInDesiredState -f $mitigationName, $mitigationValue)
             $inDesiredState = $false
         }
     }
-
-    # verify policies in Disable are in a desired state
-    foreach ( $policy in $Disable )
+    else
     {
-        if ($policy -notin $currentState.Disable)
+        # verify policies in Enable are in a desired state
+        foreach ($mitigation in $currentState.MitigationPolicy.AppConfig)
         {
-            Write-Verbose -Message ($script:localizedData.policyNotInDesiredStateDisabled -f $policy)
-            $inDesiredState = $false
+            if ($mitigation.Executable -eq $MitigationTarget)
+            {
+                if ($mitigation.$mitigationType.$mitigationName -ne $mitigationValue)
+                {
+                    Write-Verbose -Message ($script:localizedData.policyNotInDesiredState -f $mitigationName, $mitigationValue)
+                    $inDesiredState = $false
+                }
+            }
         }
     }
 
+    $mitigationMatches = [regex]::matches($currentState.MitigationPolicy.AppConfig.Executable,$MitigationTarget,"IgnoreCase")
```

Comment:
Are we using regular expression for literal matches here?  I think comparing property values with a literal comparison operator is more predictable than using regular expression.

Reply:
Updated